### PR TITLE
Using pinched grid for numerical aquifers

### PR DESCRIPTION
**Pull request by GitPaean**

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this opm-common=2389 opm-grid=517 please

**Comment by GitPaean** [comment]
jenkins build this opm-common=2389 opm-grid=517 please

**Comment by GitPaean** [comment]
closing it to reopen to see whether good luck for jenkins. It really gets stuck here.

File: ebos/eclcpgridvanguard.hh [hunk 1]
```diff
@@ -385,6 +384,23 @@ protected:
...
+        {
+            const bool has_numerical_aquifer = this->eclState().aquifer().hasNumericalAquifer();
+            int mpiSize = 1;
+            MPI_Comm_size(MPI_COMM_WORLD, &mpiSize);
+            // when there is numerical aquifers, new NNC are generated during grid processing
+            // we need to pass the NNC from root process to other processes
+            if (has_numerical_aquifer && mpiSize > 1) {
+                auto nnc_input = this->eclState().getInputNNC();
+                Opm::EclMpiSerializer ser(Dune::MPIHelper::getCollectiveCommunication());
+                ser.template broadcast(nnc_input);
```
Comment (bska) [inline ebos/eclcpgridvanguard.hh:397]: Do you really need `.template` here?  For one, there's no explicit template argument.  Secondly, I don't really see how `nnc_input` depends on a template parameter.

**Review by bska — COMMENTED** [review]
Looks good to me, I just had one small comment.  Otherwise this is mostly chasing upstream API changes.

File: ebos/eclcpgridvanguard.hh (hunk 1 above)
Comment (GitPaean) [reply ebos/eclcpgridvanguard.hh:397]: Yes. It looks like `template` is not needed. But I do not recall why it is there in the first place either. I will update it.

**Review by bska — APPROVED** [review]
Thanks for the updates.  I'll merge this once the upstream changes have been merged too.

**Comment by totto82** [comment]
> Thanks for the updates. I'll merge this once the upstream changes have been merged too.

For me it seems this and its upstreams can be merged. I plan to have the second round of the release branches ready today. Cool if this can be merged before that.

**Comment by bska** [comment]
> > Thanks for the updates. I'll merge this once the upstream changes have been merged too.
> 
> For me it seems this and its upstreams can be merged

Everything hinges on OPM/opm-common#2389 which does not have final approval yet.

**Pull request merged**